src/schema/schema_drop.c

```diff
@@ -174,31 +174,58 @@ __drop_table(WT_SESSION_IMPL *session, const char *uri, const char *cfg[])
  *     Drop a tiered store.
  */
 static int
-__drop_tiered(WT_SESSION_IMPL *session, const char *uri, const char *cfg[])
+__drop_tiered(WT_SESSION_IMPL *session, const char *uri, bool force, const char *cfg[])
 {
     WT_DATA_HANDLE *tier;
     WT_DECL_RET;
     WT_TIERED *tiered;
     u_int i;
+    const char *name;
 
+    name = NULL;
+    WT_UNUSED(cfg);
     /* Get the tiered data handle. */
     WT_RET(__wt_session_get_dhandle(session, uri, NULL, NULL, WT_DHANDLE_EXCLUSIVE));
     tiered = (WT_TIERED *)session->dhandle;
 
-    /* Drop the tiers. */
-    for (i = 0; i < WT_TIERED_MAX_TIERS; i++) {
-        tier = tiered->tiers[i].tier;
-        if (tier != NULL)
-            WT_ERR(__wt_schema_drop(session, tier->name, cfg));
+    /*
+     * We cannot remove the objects on shared storage as other systems may be accessing them too.
+     * Remove the current local object and then remove all bucket objects from the metadata only.
+     */
+    tier = tiered->tiers[WT_TIERED_INDEX_LOCAL].tier;
+    if (tier != NULL) {
+        __wt_verbose(session, WT_VERB_TIERED, "DROP_TIERED: drop local object %s", tier->name);
+        WT_WITHOUT_DHANDLE(session,
+          WT_WITH_HANDLE_LIST_WRITE_LOCK(
+            session, ret = __wt_conn_dhandle_close_all(session, tier->name, true, force)));
+        WT_ERR(__wt_metadata_remove(session, tier->name));
```

Comment:
Does this need a matching call to `wt_meta_track_drop` to ensure that the local file gets removed?

Reply:
Good catch. I also, therefore, added support for the `remove_files` configuration setting. And removed the `wt_meta_track_drop` from the object loop because we're not removing objects.